pkg/tests/schema_pulumi_test.go

```diff
@@ -59,3 +61,677 @@ outputs:
+			cloudVal:           []interface{}{},
+			programVal:         "[]",
+			// Note the difference in expected output between top level and nested properties
+			expectedOutputTopLevel: nil,
+			expectedOutputNested:   []interface{}{},
+			// Note only fails at the top level
+			expectFailTopLevel: true,
+		},
+		{
+			name:                   "set empty with planResourceChange with cloud override",
+			planResourceChange:     true,
+			schemaType:             schema.TypeSet,
+			cloudVal:               []interface{}{},
+			programVal:             "[]",
+			createCloudValOverride: true,
+			expectedOutputTopLevel: nil,
+			expectedOutputNested:   nil,
+			expectFailTopLevel:     true,
+			expectFailNested:       true,
+		},
+		{
+			name:                   "set empty without planResourceChange with cloud override",
+			planResourceChange:     false,
+			schemaType:             schema.TypeSet,
+			cloudVal:               []interface{}{},
+			programVal:             "[]",
+			createCloudValOverride: true,
+			expectedOutputTopLevel: []interface{}{},
+			expectedOutputNested:   []interface{}{},
+		},
+		{
+			name:                   "set nonempty with planResourceChange",
+			schemaType:             schema.TypeSet,
+			cloudVal:               []interface{}{"val"},
+			programVal:             `["val"]`,
+			expectedOutputTopLevel: []interface{}{"val"},
+			expectedOutputNested:   []interface{}{"val"},
+		},
+		{
+			name:                   "set nonempty without planResourceChange",
+			planResourceChange:     false,
+			schemaType:             schema.TypeSet,
+			cloudVal:               []interface{}{"val"},
+			programVal:             `["val"]`,
+			expectedOutputTopLevel: []interface{}{"val"},
+			expectedOutputNested:   []interface{}{"val"},
+		},
+		{
+			name:                   "set nonempty with planResourceChange with cloud override",
+			schemaType:             schema.TypeSet,
+			cloudVal:               []interface{}{"val"},
+			programVal:             `["val"]`,
+			createCloudValOverride: true,
+			expectedOutputTopLevel: []interface{}{"val"},
+			expectedOutputNested:   []interface{}{"val"},
+		},
+		{
+			name:                   "set nonempty without planResourceChange with cloud override",
+			planResourceChange:     false,
+			schemaType:             schema.TypeSet,
+			cloudVal:               []interface{}{"val"},
+			programVal:             `["val"]`,
+			createCloudValOverride: true,
+			expectedOutputTopLevel: []interface{}{"val"},
+			expectedOutputNested:   []interface{}{"val"},
+		},
+	} {
+		collectionPropPlural := ""
+		pluralized := tc.schemaType == schema.TypeList || tc.schemaType == schema.TypeSet
+		if pluralized {
+			collectionPropPlural += "s"
+		}
+
+		opts := []pulcheck.BridgedProviderOpt{}
+		if !tc.planResourceChange {
+			opts = append(opts, pulcheck.DisablePlanResourceChange())
+		}
+
+		t.Run(tc.name+" top level", func(t *testing.T) {
+			resMap := map[string]*schema.Resource{
```

Comment:
Nit: We are creating a nested hierarchy instead of using the existing hierarchy. 
```suggestion
		t.Run(tc.name, func(t *testing.T) {
			t.Run("top level", func(t *testing.T) {
```

Reply:
neat